IBX-5273 - Order management API (#1922)
ezsystems/developer-documentation: docs/commerce/order_management/order_management_api.md

---
description: Use PHP API to manage orders in Commerce: find, get, create and update them.
edition: commerce
---

# Order management API

!!! tip "Order management REST API"

    To learn how to manage orders with the REST API, see the [REST API reference](../../api/rest_api/rest_api_reference/rest_api_reference.html#managing-orders).

To get orders and manage them, use the `Ibexa\Contracts\OrderManagement\OrderServiceInterface` interface.

## Get single order 

### Get single order by identifier

To access a single order by using its string identifier, use the `OrderService::getOrderByIdentifier` method:

``` php
[[= include_file('code_samples/api/commerce/src/Command/OrderCommand.php', 50, 54) =]]
```

> ### Get single order by id

Reviewer: ```suggestion
### Get single order by ID
```

> To access a single order by using its numerical id, use the `OrderService::getOrder` method:

Reviewer: ```suggestion
To access a single order by using its numerical ID, use the `OrderService::getOrder` method:
```

``` php
[[= include_file('code_samples/api/commerce/src/Command/OrderCommand.php', 56, 61) =]]
```

## Get multiple orders

To fetch multiple orders, use the `OrderService::findOrders` method. 

> It follows the same search Query pattern as other APIs:

Reviewer: ```suggestion
It follows the same search query pattern as other APIs:
```

``` php
[[= include_file('code_samples/api/commerce/src/Command/OrderCommand.php', 10, 15) =]]

// ...
[[= include_file('code_samples/api/commerce/src/Command/OrderCommand.php', 79, 88) =]]
```

## Create order

> To create an order, use the `orderServiceInterface::createOrder` method and provide 

Reviewer: ```suggestion
To create an order, use the `OrderService::createOrder` method and provide 
```

it with the `Ibexa\Contracts\OrderManagement\Value\OrderCreateStruct` that contains a list of products, purchased quantities, product and total prices, as well as tax amounts.

``` php
[[= include_file('code_samples/api/commerce/src/Command/OrderCommand.php', 63, 70) =]]
```

## Update order

> You can update order after the order is created. 

Reviewer: ```suggestion
You can update an order after it is created. 
```
